validations/semantic_metadata.go

```diff
@@ -0,0 +1,116 @@
+
+		providesNode := getPropContent(modulesNode[i], providesYamlField)
+		for i, provides := range module.Provides {
+			issues = append(issues, checkMetadata(provides.Properties, provides.PropertiesMetaData, providesNode[i], mapTypeProperties)...)
+		}
+
+		requiresNode := getPropContent(modulesNode[i], requiresYamlField)
+		issues = append(issues, checkRequiresParamsAndPropertiesMetadata(requiresNode, module.Requires)...)
+
+		hooksNode := getPropContent(modulesNode[i], hooksYamlField)
+		for i, hook := range module.Hooks {
+			issues = append(issues, checkMetadata(hook.Parameters, hook.ParametersMetaData, hooksNode[i], mapTypeParameters)...)
+
+			requiresNode = getPropContent(hooksNode[i], requiresYamlField)
+			issues = append(issues, checkRequiresParamsAndPropertiesMetadata(requiresNode, hook.Requires)...)
+		}
+	}
+
+	resourcesNode := getPropContent(mtaNode, resourcesYamlField)
+	for i, resource := range mta.Resources {
+		issues = append(issues, checkMetadata(resource.Parameters, resource.ParametersMetaData, resourcesNode[i], mapTypeParameters)...)
+		issues = append(issues, checkMetadata(resource.Properties, resource.PropertiesMetaData, resourcesNode[i], mapTypeProperties)...)
+
+		requiresNode := getPropContent(resourcesNode[i], requiresYamlField)
+		issues = append(issues, checkRequiresParamsAndPropertiesMetadata(requiresNode, resource.Requires)...)
+	}
+
+	if strict {
+		return issues, nil
+	}
+	return nil, issues
+}
+
+func checkRequiresParamsAndPropertiesMetadata(requiresNodes []*yaml.Node, requiresList []mta.Requires) []YamlValidationIssue {
+	var issues []YamlValidationIssue
+
+	for i, requires := range requiresList {
+		issues = append(issues, checkMetadata(requires.Parameters, requires.ParametersMetaData, requiresNodes[i], mapTypeParameters)...)
+		issues = append(issues, checkMetadata(requires.Properties, requires.PropertiesMetaData, requiresNodes[i], mapTypeProperties)...)
+	}
+	return issues
+}
+
+// Check each property/parameter in the metadata is defined in the map, and that non-optional and non-overwritable property/parameter is not nil
+func checkMetadata(m map[string]interface{}, metadata map[string]mta.MetaData, parentNode *yaml.Node, mapType int) []YamlValidationIssue {
+	metadataNode := getPropValueByName(parentNode, mapTypes[mapType].metadataNodeName)
+	mapNode := getPropValueByName(parentNode, mapTypes[mapType].mapNodeName)
+
+	var issues []YamlValidationIssue
+	for key := range metadata {
+		_, ok := m[key]
+		if !ok {
+			keyNode := getPropByName(metadataNode, key)
+			issues = append(issues, YamlValidationIssue{Msg: fmt.Sprintf(unknownNameInMetadataMsg, key, mapTypes[mapType].entityKind), Line: keyNode.Line})
+		}
+	}
+	if metadata != nil {
+		for key, value := range m {
+			// If there's no metadata for the key we don't perform this check (since it's overwritable by default)
+			if meta, ok := metadata[key]; ok {
+				if !meta.Optional && !meta.OverWritable && value == nil {
```

Comment:
second if condition can be added to the first if

Reply:
It seemed like too many conditions for 1 if